code/datums/ai/monkey/monkey_controller.dm

```diff
@@ -0,0 +1,211 @@
+/*
+AI controllers are a datumized form of AI that simulates the input a player would otherwise give to a mob. What this means is that these datums
+have ways of interacting with a specific mob and control it.
+*/
+///OOK OOK OOK
+
+/datum/ai_controller/monkey
+	blackboard = list(BB_MONKEY_AGRESSIVE = FALSE,\
+	BB_MONKEY_BEST_FORCE_FOUND = 0,\
+	BB_MONKEY_ENEMIES = list(),\
+	BB_MONKEY_BLACKLISTITEMS = list(),\
+	BB_MONKEY_PICKUPTARGET = null,\
+	BB_MONKEY_PICKPOCKETING = FALSE,
+	BB_MONKEY_DISPOSING = FALSE,
+	BB_MONKEY_TARGET_DISPOSAL = null,
+	BB_MONKEY_CURRENT_ATTACK_TARGET = null,
+	BB_MONKEY_CURRENT_ATTACK_TARGET)
+
+/datum/ai_controller/monkey/angry
+
+/datum/ai_controller/monkey/angry/TryPossessPawn(atom/new_pawn)
+	. = ..()
+	if(. & AI_CONTROLLER_INCOMPATIBLE)
+		return
+	blackboard[BB_MONKEY_AGRESSIVE] = TRUE //Angry cunt
+
+/datum/ai_controller/monkey/TryPossessPawn(atom/new_pawn)
+	if(!isliving(new_pawn))
+		return AI_CONTROLLER_INCOMPATIBLE
+	RegisterSignal(new_pawn, COMSIG_PARENT_ATTACKBY, .proc/on_attackby)
+	RegisterSignal(new_pawn, COMSIG_ATOM_ATTACK_HAND, .proc/on_attack_hand)
+	RegisterSignal(new_pawn, COMSIG_ATOM_ATTACK_PAW, .proc/on_attack_paw)
+	RegisterSignal(new_pawn, COMSIG_ATOM_BULLET_ACT, .proc/on_bullet_act)
+	RegisterSignal(new_pawn, COMSIG_ATOM_HITBY, .proc/on_hitby)
+	RegisterSignal(new_pawn, COMSIG_MOVABLE_CROSSED, .proc/on_Crossed)
+	RegisterSignal(new_pawn, COMSIG_LIVING_START_PULL, .proc/on_startpulling)
+	RegisterSignal(new_pawn, COMSIG_LIVING_TRY_SYRINGE, .proc/on_try_syringe)
+	RegisterSignal(new_pawn, COMSIG_ATOM_HULK_ATTACK, .proc/on_attack_hulk)
+	RegisterSignal(new_pawn, COMSIG_CARBON_CUFF_ATTEMPTED, .proc/on_attempt_cuff)
+	return ..() //Run parent at end
+
+/datum/ai_controller/monkey/UnpossessPawn()
```

Comment:
Is there any reason why those are name UppercaseUppercase instead of lowercase_lowercase? Rest of procs are named that way.

Reply:
To be honest, I don't know, it seems to be a common thing here. If it's unacceptable I can change all Uppercases to lowercases or something.